pybamm_tea/tea.py

```diff
@@ -394,223 +425,191 @@ def print_capacities_and_potentials(self):
+            # calculate lithium in positive electrode at SOC = 1
+            y100 = ((pava.get("Lithium inventory [mA.h.cm-2]") - n_ne_100)
+                      / (pava.get("Maximum concentration in positive electrode [mol.m-3]")
+                         * pava.get("Positive electrode thickness [m]")
+                         * pava.get("Positive electrode active material volume fraction")
+                         * 96485 / 3.6 / 10000))
+            if y0 < 0 or y100 < 0 or y0 > 1 or y100 > 1:
+                raise ValueError("Error: Stoichiometry calculation for positive electrode failed.")
+            if pava.get("Negative electrode OCP [V]") is not None and pava.get("Positive electrode OCP [V]") is not None:
+                pava["Lower voltage cut-off [V]"] = pava.get("Positive electrode OCP [V]")(y0).evaluate() - pava.get("Negative electrode OCP [V]")(x0).evaluate()
+                pava["Upper voltage cut-off [V]"] = pava.get("Positive electrode OCP [V]")(y100).evaluate() - pava.get("Negative electrode OCP [V]")(x100).evaluate()
+            if pava.get("Negative electrode OCP [V]") is None and pava.get("Positive electrode OCP [V]") is not None:
+                pava["Lower voltage cut-off [V]"] = pava.get("Positive electrode OCP [V]")(y0).evaluate() - pava.get("Negative electrode average OCP [V]")
+                pava["Upper voltage cut-off [V]"] = pava.get("Positive electrode OCP [V]")(y100).evaluate() - pava.get("Negative electrode average OCP [V]")
+            if pava.get("Negative electrode OCP [V]") is not None and pava.get("Positive electrode OCP [V]") is None:
+                pava["Lower voltage cut-off [V]"] = pava.get("Positive electrode average OCP [V]") - pava.get("Negative electrode OCP [V]")(x0).evaluate()
+                pava["Upper voltage cut-off [V]"] = pava.get("Positive electrode OCP average [V]") - pava.get("Negative electrode OCP [V]")(x100).evaluate()
         elif (
             pava.get("Negative electrode OCP [V]") is not None
             and pava.get("Positive electrode OCP [V]") is not None
             and pava.get("Lower voltage cut-off [V]") is not None
             and pava.get("Upper voltage cut-off [V]") is not None
         ):
+            # stoichiometries at 0 and 100% SOC based on cell potential limits
             x0, x100, y100, y0 = pybamm.lithium_ion.get_min_max_stoichiometries(
                 pybamm.ParameterValues(pava)
-            )  # stoichiometries at 0 and 100% SOC based on cell potential limits
+            )
         else:
             raise ValueError("Error: Stoichiometry calculation failed.")
         stack_ed["Negative electrode stoichiometry at 0% SoC"] = x0
         stack_ed["Negative electrode stoichiometry at 100% SoC"] = x100
         stack_ed["Positive electrode stoichiometry at 100% SoC"] = y100
         stack_ed["Positive electrode stoichiometry at 0% SoC"] = y0
 
+        # update initial concentrations in electrodes to SoC = 1
```

Comment:
is this consistent wit previous calculations?

it seems like there are multiple places where the same values get calculated, and it is confusing